classes/profile.php

```diff
@@ -97,16 +97,78 @@ public static function getashar($id): string {
     }
 
     /**
-     * Gets the script type of th request.
+     * @return int
+     * @throws \dml_exception
+     */
+    public static function getnumautoprofiles(): int {
+        global $DB;
+        return $DB->get_field_sql(
+            "SELECT count(*) FROM {tool_excimer_profiles} WHERE logmethod = ?",
+            [ manager::LOGMETHOD_AUTO ]
+        );
+    }
+
+    /**
+     * @return object
+     * @throws \dml_exception
+     */
+    public static function getfastestautoprofile(): object {
+        global $DB;
+        $sql = "
+            SELECT id, duration
+            FROM {tool_excimer_profiles}
```

Comment:
please indent the left capital works so their right letters are aligned, eg

https://docs.moodle.org/dev/SQL_coding_style#Indentation

Reply:
This has been done.